# **UbaidUllah**

## **BYTEWISE FELLOWSHIP**

### **Task 3: Set up Jupiter Notebook, explore it and see how you can add fancy heading to it**

---

## **Introduction**
In this task, we will get familiar with the three types of cells in Jupyter Notebook: **Code**, **Markdown**, and **Raw**. We'll also explore advanced features and functionalities.

---

## **1. Code**

Code cells are the primary cells for writing and running code in Jupyter Notebook. When executed, the code runs in the kernel and the output is displayed directly below the cell.
### Coding Rules
 
- Follow Python's PEP 8 style guidelines.
- Use meaningful variable and function names.
- Comment your code to explain decisions.
- Handle errors using try-except blocks.
- Avoid hardcoding values; use constants.
- Modularize code into functions or classes.
- Document functions and classes with docstrings.
- Test code thoroughly under different scenarios.
- Use version control (e.g., Git) for tracking changes.



In [8]:
# Various Data Types
a = 10
print("Integer:", a)
b = 3.14
print("Float:", b)
c = 1 + 2j
print("Complex Number:", c)
name = "Ubaid Ullah"
print("String:", name)
fruits = ["apple", "banana", "cherry"]
print("List:", fruits)
colors = ("red", "green", "blue")
print("Tuple:", colors)
student = {"name": "Tehreem", "age": 21, "course": "Data Science"}
print("Dictionary:", student)
unique_numbers = {1, 2, 3, 4, 5}
print("Set:", unique_numbers)
is_student = True
print("Boolean:", is_student)

Integer: 10
Float: 3.14
Complex Number: (1+2j)
String: Ubaid Ullah
List: ['apple', 'banana', 'cherry']
Tuple: ('red', 'green', 'blue')
Dictionary: {'name': 'Tehreem', 'age': 21, 'course': 'Data Science'}
Set: {1, 2, 3, 4, 5}
Boolean: True


In [9]:
# Import the math module for mathematical operations
import math  

# Define constants
MAX_RETRIES = 3

# Function to calculate the area of a circle given its radius
def calculate_circle_area(radius):
    """
    Calculate the area of a circle.

    Args:
        radius (float): The radius of the circle.

    Returns:
        float: The area of the circle.
    """
    area = math.pi * radius ** 2  # Formula 
    return area


radius = 5.0  # Example radius

# Calculate area of the circle using the function
area = calculate_circle_area(radius)

# Print the calculated area
print(f"The area of a circle with radius {radius} is: {area:.2f}")


The area of a circle with radius 5.0 is: 78.54


## **2. MARKDOWN**

Markdown is a popular markup language that is the superset of HTML. Jupyter Notebook supports Markdown, allowing for easy formatting of text and integration with code and other content.

# Adding Headers, Subheadings, and Sub-subheadings:

## Heading 1
- Subheading 1.1
  - Sub-subheading 1.1.1

## Heading 2
- Subheading 2.1
  - Sub-subheading 2.1.1
  - Sub-subheading 2.1.2
- Subheading 2.2
  - Sub-subheading 2.2.1

## Heading 3
- Subheading 3.1
  - Sub-subheading 3.1.1
  - Sub-subheading 3.1.2
- Subheading 3.2
  - Sub-subheading 3.2.1

## Heading 4
- Subheading 4.1
  - Sub-subheading 4.1.1
  - Sub-subheading 4.1.2
- Subheading 4.2
  - Sub-subheading 4.2.1
 


## Text Formatting

Markdown supports various text formatting options:
- **Bold**: Use double asterisks (`**bold text**`) or double underscores (`__bold text__`).
- *Italic*: Use single asterisks (`*italic text*`) or single underscores (`_italic text_`).
- ~~Strikethrough~~: Use double tildes (`~~strikethrough text~~`).
- `Inline Code`: Use backticks (`inline code`).

## Ordered List
1. Item 1
2. Item 2

## Unordered List
- Bullet Point 1
- Bullet Point 2
  - Sub-bullet Point 2.1
  - Sub-bullet Point 2.2

## Links and Images

### Inserting an Image with a Link
[![Python Tools for Developers](https://www.turing.com/kb/wp-content/uploads/2023/03/Top-10-Python-Tools-for-Developers-Featured-Image.png)](https://www.turing.com/kb/top-10-python-tools-for-developers)

### Inserting a Website Link
[Python Programming at W3Schools](https://www.w3schools.com/python/)


# Cretion of table 





## Table

| Name         | Position       | Salary ($) |
|--------------|----------------|------------|
| John Doe     | Developer      | 50000      |
| Jane Smith   | Designer       | 45000      |
| Michael Brown| Data Analyst   | 60000      |


# Mathematical Formulas

## Introduction

This section demonstrates LaTeX equations for mathematical expressions.

## Equations

### Equation 1: Quadratic Formula

The quadratic formula is given by:

$$
x = \frac{{-b \pm \sqrt{{b^2 - 4ac}}}}{{2a}}
$$

### Equation 2: Pythagorean Theorem

The Pythagorean theorem states:

$$
c = \sqrt{{a^2 + b^2}}
$$


## Blockquote

> "The greatest glory in living lies not in never falling, but in rising every time we fall."
>
> \- Nelson Mandela

### Explanation:
- **Heading 1 (`# Blockquotes Example`)**: Introduces the main topic of the document.
- **Heading 2 (`## Introduction`)**: Provides a subsection that explains the purpose of blockquotes.
- **Blockquote (`> "The greatest glory ... we fall."`)**: Inserts a blockquote using the `>` symbol followed by the quoted text and attribution.

### Result:
When rendered in Markdown, the document will display as:

---

# Blockquotes Example

## Introduction

Blockquotes are used to emphasize quoted text or citations from external sources.

## Blockquote

> "The greatest glory in living lies not in never falling, but in rising every time we fall."
>
> \- Nelson Mandela

---

Blockquotes are commonly used to provide context, cite sources, or emphasize important statements within Markdown documents. Adjust the quoted text and attribution as needed for your specific content.


# Endnotes and Citations 



## Content

Here is some text with a footnote[^1] and a citation[^2].

[^1]: This is a footnote explaining additional information.

[^2]: Author, "Title of the Source," Publisher, Year.




# Task List Example


- [x] Task 1: Complete Markdown tasks
- [ ] Task 2: Review advanced features
- [ ] Task 3: Explore additional Markdown capabilities


# Table of Contents 

## Table of Contents

1. [Introduction](#introduction)
2. [Features](#features)
   1. [Advanced Markdown Features](#advanced-markdown-features)
   2. [More Advanced Features](#more-advanced-features)
3. [Examples](#examples)
4. [Conclusion](#conclusion)


### 3. Raw Cell

A raw cell in Jupyter Notebook is a type of cell that allows you to input content in various formats without it being executed as code or processed as Markdown. Raw cells are particularly useful for displaying content that you want to appear exactly as entered, such as raw HTML, LaTeX equations, scripts, or other formats that should not be interpreted by the notebook's kernel.

Raw cells are different from Markdown cells in that they do not render Markdown syntax or execute code. Instead, they display the content directly as it is written, maintaining its original formatting and structure.

Next, we'll proceed to provide examples of different types of content that can be included in a raw cell.
